components/collections/src/codepointtrie/cptrie.rs

```diff
@@ -384,7 +384,7 @@ impl<'trie, T: TrieValue> CodePointTrie<'trie, T> {
             } else {
                 return self.trie_error_val_index();
             };
-        let fast_index_val: u32 = index_array_val as u32 + (code_point & FAST_TYPE_DATA_MASK);
+        let fast_index_val: u32 = w!((index_array_val as u32) + code_point & FAST_TYPE_DATA_MASK);
```

Comment:
please revert the parentheses, I shouldn't need to know the precedence of `+`/`&`

Reply:
+1 especially if the macro results in a different precedence

Reply:
The parens are necessary for the macro to parse this.

`foo as bar` is multiple tts.

Reply:
Oh, you want additional parens, I see. I can add.

Reply:
I want no more parens than there were before

Reply:
That was due to the macro parsing limitations, not precedence (the macro refuses to parse `foo as bar` as a single tt).

But I split it out into a variable instead.

Reply:
Sorry that was phrased ambiguously, I was referring to your comment that called the parens that you had removed and I was asking you to add back "additional". I'm fine have more parens than before if the macro requires it. I'm not fine removing parens that require the reader to know the precedence of `+` and `&`.